main/acle.md

```diff
@@ -4571,6 +4595,56 @@ two pointers, ignoring the tags.
 The return value is the sign-extended result of the computation.
 The tag bits in the input pointers are ignored for this operation.
 
+# Guarded Control Stack intrinsics
+
+## Introduction
+
+This section describes the intrinsics for the instructions of the
+Guarded Control Stack (GCS) extension. The GCS instructions are present
+in the AArch64 execution state only.
+
+When GCS protection is enabled then function calls save the return
+address to a separate stack, the GCS, that is checked against the actual
+return address when the function returns. At runtime GCS protection can
+be disabled and then calls and returns do not access the GCS. The GCS
+grows down and a GCS pointer points to the last entry of the GCS.
+Each thread has a separate GCS and GCS pointer.
+
+To use the intrinsics, `arm_acle.h` needs to be included.
+
+The intrinsics are only valid to call when GCS instructions are
+supported. The `__chkfeat` intrinsics with `_CHKFEAT_GCS` can be used
+to check if GCS protection is enabled at runtime. If GCS protection is
+enabled then GCS instructions are supported and the code was compiled
+in a GCS protection compatible way (`__ARM_FEATURE_GCS_DEFAULT` was
+defined).
```

Comment:
The final sentence here is a bit weird and I'm not sure what exactly it's saying. I _think_ what it's saying is that "GCS enabled implies GCS available", i.e. if you have ``if (__chkfeat(_CHKFEAT_GCS)) { }`` then you know GCS must be available inside that block, or perhaps the other way around that if you know GCS isn't available then you know ``__chkfeat(_CHKFEAT_GCS)`` will never return true.

Reply:
yeah this should be worded better, or removed.

i was trying to point out that the runtime will only enable GCS if all code is built with GCS support, so if __chkfeat check returns true, that implies __ARM_FEATURE_GCS_DEFAULT, no need to check that separately.